src/dynarec/rv64/dynarec_rv64_emit_tests.c

```diff
@@ -23,6 +23,69 @@
 #include "dynarec_rv64_functions.h"
 #include "dynarec_rv64_helper.h"
 
+// emit CMP8 instruction, from cmp s1, s2, using s3, s4, s5 and s6 as scratch
+void emit_cmp8(dynarec_rv64_t* dyn, int ninst, int s1, int s2, int s3, int s4, int s5, int s6) {
+    CLEAR_FLAGS();
+    IFX_PENDOR0 {
+        SB(s1, xEmu, offsetof(x64emu_t, op1));
+        SB(s2, xEmu, offsetof(x64emu_t, op2));
+        SET_DF(s4, d_cmp8);
+    } else {
+        SET_DFNONE(s4);
+    }
+
+    IFX(X_AF | X_CF | X_OF) {
+        // for later flag calculation
+        NOT(s5, s1);
+    }
+
+    // It's a cmp, we can't store the result back to s1.
+    SUB(s6, s1, s2);
+    IFX_PENDOR0 {
+        SB(s6, xEmu, offsetof(x64emu_t, res));
+    }
+    IFX(X_SF) {
+        SRLI(s3, s6, 7);
+        BEQZ(s3, 4);
+        ORI(xFlags, xFlags, 1 << F_SF);
+    }
+    CALC_SUB_FLAGS(s5, s2, s6, s3, s4, 8);
+    IFX(X_ZF) {
+        BEQZ(s6, 4);
+        ORI(xFlags, xFlags, 1 << F_ZF);
+    }
+    IFX(X_PF) {
+        emit_pf(dyn, ninst, s6, s3, s4);
+    }
+}
+
+// emit CMP8 instruction, from cmp s1 , 0, using s3 and s4 as scratch
+void emit_cmp8_0(dynarec_rv64_t* dyn, int ninst, int s1, int s3, int s4) {
+    CLEAR_FLAGS();
+    IFX_PENDOR0 {
+        SB(s1, xEmu, offsetof(x64emu_t, op1));
+        ADDI(s4, xZR, 0);
```

Comment:
we probably don't need `s4` here and can use `xZR` all along

Reply:
Yes, we don't need `s4` here, I guess we can just `SB(xZR, xEmu, offsetof(x64emu_t, op2));`? `s4` is indeed written in `emit_pf`, but that shouldn't matter?

Reply:
Yes. It was just to also remove it completly from the function call, but we have to keep it for emit_pf...

Reply:
Force-pushed.